### DISCUSSION TITLE
Is there any option in tiny-curl to disable extracting and validating AuthorizationKeyIdentifier

### DISCUSSION THREAD

**Discussion by BhanuPrakash-P** [curl/curl#14226] [Q&A] [answered]
We have an application which uses tiny-curl, mbedtls for SSL connections.
On our ESP-IDF v5.0.1, SSL connections were fine and able to communicate with Azure IoT.
Recently, we started upgrading ESP-IDF to v5.1.2 in which mbedtls has added length validations for AuthorizationKeyIdentifier and observed an error as follows.
_Curl Debug Callback()[0]: *  Error reading private key (nil) - mbedTLS: (-0x2566) X509 - The extension tag or value is invalid : ASN1 - Actual length differs from expected length_

Is there anyway or option in tiny-curl to disable validation of AuthorizationKeyIdentifier?

Changes in mbedTLS causing the error:
https://github.com/Mbed-TLS/mbedtls/commit/6ec839a1f95c5270eb5347a675b4c7497638737c#diff-fe0507636920d9b43e948fb12d2850460e8cbe2b7d7c56278def07ea1ba41c6f 

Any pointers and inputs are highly appreciated.

**Comment by bagder** [comment, marked as answer]
That seems like an error returned from an mbedTLS call. Changes for that treatment should then presumably be done in mbedTLS, shouldn't they?

Or are you asking if curl can be made to ignore a specific return code?

**Reply by BhanuPrakash-P** [reply]
Though the ability to register a callback in curl would be a good option, this needs to be handled in mbedTLS. 

Thanks.
